src/components/shared/Button/Button.tsx

```diff
@@ -0,0 +1,58 @@
+import clsx from "clsx";
+import Link from "next/link";
+import ButtonIcon from "./ButtonIcon";
+import { IconName } from "@/types";
+
+export interface ButtonProps {
+  variant: "primary" | "inversion" | "secondary";
+  size: "medium" | "large";
+  icon: boolean;
+  iconName?: IconName;
+  label: string;
+  href: string;
+  ariaDescription: string;
+  disabled?: boolean;
+}
+
+export const Button = ({
+  variant,
+  label,
+  size,
+  href,
+  icon,
+  iconName,
+  ariaDescription,
+  disabled = false,
+}: ButtonProps) => {
+  const variants = {
+    primary:
+      "contrast:bg-black00 contrast:text-yellow bg-redMain text-backgroundMain desktop:hover:bg-[#7A0003] active:bg-redMain",
+    inversion:
+      "constrast:bg-yellow constrast:desktop:hover:bg-[#DAD01C] contrast:text-black00 bg-backgroundMain text-redMain desktop:hover:bg-[#F0EFEB] active:bg-backgroundMain",
+    secondary:
+      "constrast:bg-yellow constrast:desktop:hover:bg-[#DAD01C] contrast:text-black00 bg-greenLight text-black desktop:hover:bg-[#BDD2BC] active:bg-greenLight",
+  };
+
+  const sizes = {
+    medium: "text-16",
+    large: "text-lg",
+  };
+  return (
+    <Link
+      href={disabled ? "#" : href}
```

Comment:
Setting `href="#"` for disabled links is not ideal. You can use `role="button"` and remove the `href` altogether if the link is disabled